compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3464,7 +3464,9 @@ public void visit(BLangErrorConstructorExpr errorConstructorExpr, AnalyzerData d
 
         List<BType> errorDetailTypes = new ArrayList<>(expandedCandidates.size());
         for (BType expandedCandidate : expandedCandidates) {
-            BType detailType = ((BErrorType) Types.getReferredType(expandedCandidate)).detailType;
+            BType detailType = Types.getReferredType(expandedCandidate);
+            detailType = Types.getEffectiveType(detailType);
+            detailType = ((BErrorType) detailType).detailType;
```

Comment:
```java
BType detailType = ((BErrorType) Types.getEffectiveType(Types.getReferredType(expandedCandidate))).detailType;
```

Reply:
This approach decrease the readability of the code ryt?
wdyt?

Reply:
No, IMO here the refactored version eliminates some unnecessary variables assignments which can make the code easier to follow. (we can see more clearly the sequence of operations being performed on the `expandedCandidate` variable.)

Reply:
ok, Will update

Reply:
Updated